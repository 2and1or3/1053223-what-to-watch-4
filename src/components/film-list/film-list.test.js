import React from "react";
import renderer from "react-test-renderer";

import FilmList from './film-list.jsx';

const films = [
  {
    id: `0`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    release: `2014`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    voiceCount: 240,
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  },
  {
    id: `1`,
    title: `the Grand Budapest Hotel`,
    preview: `img/bohemian-rhapsody.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    release: `2014`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    voiceCount: 240,
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  },
  {
    id: `2`,
    title: `Macbeth`,
    preview: `img/macbeth.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    release: `2014`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    voiceCount: 240,
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  }];

it(`Render FilmList component`, () => {
  const tree = renderer
    .create(<FilmList films = {films} onCardClick = {() => {}}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
