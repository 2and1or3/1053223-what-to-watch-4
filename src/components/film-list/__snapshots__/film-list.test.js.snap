// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FilmList component Render full FilmList component 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog "
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      className="catalog__genres-list"
    >
      <li
        className="catalog__genres-item catalog__genres-item--active"
        id="all"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          All genres
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="comedy"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Comedies
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="crime"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Crime
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="documentary"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Documentary
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="drama"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Dramas
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="horror"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Horror
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="family"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Kids & Family
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="romance"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Romance
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="sciFi"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Sci-Fi
        </a>
      </li>
      <li
        className="catalog__genres-item "
        id="thriller"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Thrillers
        </a>
      </li>
    </ul>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            the Grand Budapest Hotel
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
    </div>
    <div
      className="catalog__more"
    >
      <button
        className="catalog__button"
        onClick={[Function]}
        type="button"
      >
        Show more
      </button>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Render FilmList component Render short FilmList component 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog catalog--like-this"
  >
    <h2
      className="catalog__title "
    >
      More like this
    </h2>
    
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            the Grand Budapest Hotel
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
          onClick={[Function]}
        >
          <video
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
    </div>
    
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
